### ISSUE TITLE
training runtime is referenced to a ClusterTrainingRuntime although the runtime specified in `train()` is namespace-scoped

### ISSUE THREAD

**Issue by dpnam2112** [kubeflow/sdk#536]
### What happened?

I created a namespace-scoped training runtime in a namespace
```yaml
apiVersion: trainer.kubeflow.org/v1alpha1
kind: TrainingRuntime
metadata:
  name: zso-torch-distributed
  namespace: hpc-zso
  labels:
    trainer.kubeflow.org/framework: torch
spec:
...
```
When I submit the job in the namespace using the SDK and use the above runtime as the training runtime for the job:
```python
from kubeflow.trainer import KubernetesBackendConfig, TrainerClient, CustomTrainer
import time
import logging

logging.basicConfig(level=logging.DEBUG)

k8s_config = KubernetesBackendConfig(
    namespace="hpc-zso",
    config_file="~/kubeconfig.yaml"
)
client = TrainerClient(backend_config=k8s_config)

def train_fn():
    import time
    time.sleep(300)

print("Submitting TrainJob...")

job_id = client.train(
    runtime="zso-torch-distributed",
    trainer=CustomTrainer(
        image="pytorch/pytorch:2.8.0-cuda12.8-cudnn9-devel",
        func=train_fn,
        num_nodes=1,
        resources_per_node={
            "cpu": "2",
            "memory": "4Gi"
        },
    )
)

print(f"Job submitted: {job_id}") 
```
The SDK throws this error

<img width="2560" height="111" alt="Image" src="https://github.com/user-attachments/assets/7d7b65cd-8acb-4c5d-a38c-e9b852d25cf0" />

Although before emitting this error log, it does find a trainingruntime in this namespace

<img width="2034" height="147" alt="Image" src="https://github.com/user-attachments/assets/156e509f-91e8-427c-aa21-4ae7036daab6" />

### What did you expect to happen?
if there is a `TrainingRuntime` with that name exists  in the namespace where the job is created, the referenced runtime should be the namespace-scoped one instead of the `ClusterTrainingRuntime` with the same name.

### Fix suggestion

In this [line](https://github.com/kubeflow/sdk/blob/main/kubeflow/trainer/backends/kubernetes/backend.py#L806),
the logic should pass the runtime kind, either `ClusterTrainingRuntime` or `TrainingRuntime`, to the instance `models.TrainerV1alpha1RuntimeRef`.

### Environment

Kubernetes version:
```bash
kubectl version
Warning: Use tokens from the TokenRequest API or manually created secret-based tokens instead of auto-generated secret-based tokens.
Client Version: v1.34.1
Kustomize Version: v5.7.1
Server Version: v1.34.6+rke2r3
```
Kubeflow Trainer version: `kubeflow/trainer/trainer-controller-manager:v2.2.1`

Kubeflow Python SDK version: 0.4.1

### Impacted by this bug?

Give it a 👍 We prioritize the issues with most 👍

**Comment by andreyvelich** [comment]
Thanks for creating this @dpnam2112!
@shaikmoeed Please can you take a look? I guess, in this PR we didn't update the `train()` API to ensure the namespace-scoped TrainingRuntime is used: https://github.com/kubeflow/sdk/pull/130
cc @abhijeet-dhumal @Fiona-Waters @kramaranya @szaher 

/remove-label needs-triage
/area trainer

**Comment by Sumitha007** [comment]
Hi! I'd like to work on this issue. I'll reproduce the issue locally and investigate a fix. Thanks!

**Comment by andreyvelich** [comment]
That would be great, thank you @Sumitha007!
/assign @Sumitha007

**Comment by Sumitha007** [comment]
Hi @andreyvelich, thanks for assigning this issue. From my understanding, the changes in #549 now propagate Runtime.kind to TrainJob.spec.runtimeRef.kind, which appears to address the behavior described here. Could you please confirm whether this issue will be fully resolved by #549, or if there are any remaining cases that you'd like me to address? Thanks
